include/stl2/view/drop.hpp

```diff
@@ -0,0 +1,228 @@
+#include <stl2/detail/semiregular_box.hpp>
+#include <stl2/detail/iterator/concepts.hpp>
+#include <stl2/detail/functional/invoke.hpp>
+#include <stl2/detail/range/access.hpp>
+#include <stl2/detail/range/concepts.hpp>
+#include <stl2/detail/view/view_closure.hpp>
+#include <stl2/view/all.hpp>
+#include <stl2/view/view_interface.hpp>
+
+STL2_OPEN_NAMESPACE {
+	namespace ext {
+		template <Range R>
+		requires
+			View<R>
+		struct drop_view;
+
+		template <Range R>
+		requires
+			View<R>
+		class __drop_view : public view_interface<drop_view<R>> {
+			using D = iter_difference_t<iterator_t<R>>;
+
+			template <bool Const>
+			struct __sentinel;
+		public:
+			__drop_view() = default;
+
+			constexpr __drop_view(R base, D count)
+				: base_(std::move(base)),
+				  count_(count)
+			{}
+
+			template <Range O>
+			requires
+				ext::ViewableRange<O> &&
+				_ConstructibleFromRange<R, O>
+			constexpr __drop_view(O&& o, D count)
+				: base_(view::all(std::forward<O>(o))),
+				  count_(count)
+			{}
+
+			constexpr R base() const
+			{
+				return base_;
+			}
+
+			constexpr auto begin()
+			{
+				return begin_impl(*this);
+			}
+
+			constexpr auto begin() const
+			requires
+				Range<R const>
+			{
+				return begin_impl(*this);
+			}
+
+			constexpr auto begin()
+			requires
+				SizedRange<R>
+			{
+				return begin_impl(*this);
+			}
+
+			constexpr auto begin() const
+			requires
+				SizedRange<R const>
+			{
+				return begin_impl(*this);
+			}
+
+			constexpr auto end()
+			{
+				return __sentinel<SimpleView<R>>(__stl2::end(base_));
+			}
+
+			constexpr auto end() const
+			requires
+				Range<R const>
+			{
+				return __sentinel<true>(__stl2::end(base_));
+			}
+
+			constexpr auto size() const
+			requires
+				SizedRange<R>
+			{
+				return __stl2::distance(base_) < count_ ? 0 : __stl2::distance(base_) - count_;
+			}
+		private:
+			R base_;
+			D count_;
+
+			template <class X>
+			static constexpr decltype(auto) begin_impl(X&& x)
+			{
+				return __stl2::next(__stl2::begin(x.base_), x.count_);
+			}
+
+			template <class X>
+			static constexpr decltype(auto) begin_impl(X&& x)
+			requires
+				SizedRange<R>
+			{
+				D const dist = __stl2::distance(x.base_);
+				return (__stl2::next(__stl2::begin(x.base_), dist < x.count_ ? dist : x.count_));
+			}
+		};
+
+		template <class R>
+		template <bool Const>
+		class __drop_view<R>::__sentinel {
+			using Base = __maybe_const<Const, R>;
+			sentinel_t<Base> end_ {};
+		public:
+			__sentinel() = default;
+
+			constexpr explicit __sentinel(sentinel_t<Base> end)
+				: end_(end)
+			{}
+
+			constexpr __sentinel(__sentinel<!Const> s)
+			requires
+				Const &&
+				ConvertibleTo<sentinel_t<R>, sentinel_t<Base>>
+				: end_(s.base())
+			{}
+
+			constexpr sentinel_t<Base> base() const
+			{
+				return end_;
+			}
+
+			friend constexpr bool operator==(__sentinel const& x, iterator_t<Base> const& y)
+			requires
+				EqualityComparable<iterator_t<Base>>
```

Comment:
It doesn't matter if we axe this custom sentinel, but FYI, these `EqualityComparable` requirements are bogus: there are no iterators being compared here. I think @ericniebler put them in `take_view` just to see if I was paying attention.